src/main/java/org/datadog/jenkins/plugins/datadog/clients/DatadogHttpClient.java

```diff
@@ -124,6 +117,37 @@ private DatadogHttpClient(String url, String logIntakeUrl, Secret apiKey) {
         this.logIntakeUrl = logIntakeUrl;
     }
 
+    public static void validateCongiguration(String url, String logIntakeUrl, Secret apiKey) throws RuntimeException, IOException {
+        if (url == null || url.isEmpty() || !validateTargetURL(url)) {
+            throw new RuntimeException("Datadog Target URL is not set properly");
+        }
+        if (apiKey == null || Secret.toString(apiKey).isEmpty()){
+            throw new RuntimeException("Datadog API Key is not set properly");
+        }
+        if (validateTargetURL(url)) {
+            throw new RuntimeException("Datadog Target URL must be configured in the form <http|https>://<url>/");
+        }
+        if (DatadogHttpClient.isCollectBuildLogEnabled() && (logIntakeUrl == null || logIntakeUrl.isEmpty())){
+            logger.warning("Datadog Log Intake URL is not set properly");
+        }
+        if (DatadogHttpClient.isCollectBuildLogEnabled() && !validateTargetURL(logIntakeUrl)) {
+            logger.warning("Datadog Target Log Intake URL must be configured in the form <http|https>://<url>/");
+        }
+        if (!validateDefaultIntakeConnection(url, apiKey)) {
+            instance.setDefaultIntakeConnectionBroken(true);
+            logger.severe("Connection broken, please double check both your API URL and Key");
+        }
+        if (!validateLogIntakeConnection(logIntakeUrl, apiKey)) {
+            instance.setLogIntakeConnectionBroken(true);
+            logger.severe("Connection broken, please double check both your Log Intake URL and Key");
+        }
+        return;
```

Comment:
So validation will pass even though both the url and the logIntakeUrl are invalid (or the apiKey is invalid).
Maybe we should at least fail it metrics can't be submitted?

Also to come back to the UI part, we should be able to display a FormWarning if the plugin can't connect.

Reply:
Oh that's a good idea. I agree there are more validations that can be performed.